src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Map.cs

```diff
@@ -0,0 +1,271 @@
+        /// </summary>
+        /// <param name="_params">Parameters.</param>
+        public Map(params object[] _params) : base(Runtime.New<Map>(_params))
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the Map class.
+        /// </summary>
+        /// <param name="jsHandle">Js handle.</param>
+        internal Map(IntPtr jsHandle) : base(jsHandle)
+        { }
+
+        /// <summary>
+        /// Gets a value indicating whether the Map object has a fixed size.
+        /// </summary>
+        public bool IsFixedSize => false;
+
+        /// <summary>
+        /// Gets a value indicating whether the Map object is read-only.
+        /// </summary>
+        public bool IsReadOnly => false;
+
+        /// <summary>
+        /// Gets an System.Collections.ICollection object containing the keys of the Map object.
+        /// </summary>
+        public ICollection Keys => new MapItemCollection(this, "keys");
+
+        /// <summary>
+        /// Gets an System.Collections.ICollection object containing the values of the Map object.
+        /// </summary>
+        public ICollection Values => new MapItemCollection(this, "values");
+
+        public int Count => (int)GetObjectProperty("size");
+
+        public bool IsSynchronized => false;
+
+        public object SyncRoot => false;
+
+        public void Add(object key, object? value) => Invoke("set", key, value);
+
+        public void Clear() => Invoke("clear");
+
+        public bool Contains(object key) => (bool)Invoke("has", key);
+
+        public IDictionaryEnumerator GetEnumerator() => new MapEnumerator(this);
+
+        public void Remove(object key) => Invoke("delete", key);
+
+        public void CopyTo(System.Array array, int index) => throw new NotImplementedException();
+
+        // Construct and return an enumerator.
+        IEnumerator IEnumerable.GetEnumerator() => new MapEnumerator(this);
+
+        /// <summary>
+        /// Gets or sets the Map with the key specified by <paramref name="key" />.
+        /// </summary>
+        /// <param name="key">The key.</param>
+        public object? this[object key]
+        {
+            get
+            {
+                return Invoke("get", key);
+            }
+            set
+            {
+                Invoke("set", key, value);
+            }
+        }
+
+        private sealed class MapEnumerator : IDictionaryEnumerator, IDisposable
+        {
+            private JSObject? _mapIterator;
+            private readonly Map _map;
+            public MapEnumerator(Map map)
+            {
+                _map = map;
+            }
+
+            // Return the current item.
+            public object Current => new DictionaryEntry(Key, Value);
+
+            // Return the current dictionary entry.
+            public DictionaryEntry Entry => (DictionaryEntry)Current;
+
+            // Return the key of the current item.
+            public object Key { get; private set; } = new object();
```

Comment:
Seems a little strange to be assigning a new object to Key here.  Is that intentional?